### DISCUSSION TITLE
email notification configuration

### DISCUSSION THREAD

**Discussion by Stamka** [DefectDojo/django-DefectDojo#7055] [Q&A] [answered]
Hello everyone. 
I tried to find manual for integrate DD with email notifications and find nothing except some discussion which dont reveal any information.
I cant do it without your help.

How i launch DD/
add to ./docker/environments/postgres-redis.env
DD_EMAIL_URL=smtp://<username>:<password>@<smtp_domain>:587

./dc-up.sh postgres-redis 

in settings/system_settings add send email to parameter
in settings/notification turn on ALL email notification

then I manually create products engagements findings and get nothing

please help

**Comment by Stamka** [comment]
Logs with exception
celeryworker_1  | OSError: [Errno 99] Cannot assign requested address
celeryworker_1  | [31/Oct/2022 13:54:47] INFO [celery.app.trace:131] Task dojo.notifications.helper.send_mail_notification[3504ce1b-fe24-4ef8-9c31-9ba04bc7d7f8] succeeded in 0.021593709010630846s: None
celeryworker_1  | [31/Oct/2022 13:54:47] INFO [celery.worker.strategy:161] Task dojo.notifications.helper.send_mail_notification[6b8b1f4d-d3fc-4462-a63a-b22acb49efc4] received
celeryworker_1  | [31/Oct/2022 13:54:47] ERROR [dojo.notifications.helper:298] [Errno 99] Cannot assign requested address
celeryworker_1  | Traceback (most recent call last):
celeryworker_1  |   File "/app/dojo/notifications/helper.py", line 295, in send_mail_notification
celeryworker_1  |     email.send(fail_silently=False)
celeryworker_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/mail/message.py", line 284, in send
celeryworker_1  |     return self.get_connection(fail_silently).send_messages([self])
celeryworker_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/mail/backends/smtp.py", line 102, in send_messages
celeryworker_1  |     new_conn_created = self.open()
celeryworker_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/mail/backends/smtp.py", line 62, in open
celeryworker_1  |     self.connection = self.connection_class(self.host, self.port, **connection_params)
celeryworker_1  |   File "/usr/local/lib/python3.8/smtplib.py", line 255, in __init__
celeryworker_1  |     (code, msg) = self.connect(host, port)
celeryworker_1  |   File "/usr/local/lib/python3.8/smtplib.py", line 339, in connect
celeryworker_1  |     self.sock = self._get_socket(host, port, self.timeout)
celeryworker_1  |   File "/usr/local/lib/python3.8/smtplib.py", line 310, in _get_socket
celeryworker_1  |     return socket.create_connection((host, port), timeout,
celeryworker_1  |   File "/usr/local/lib/python3.8/socket.py", line 808, in create_connection
celeryworker_1  |     raise err
celeryworker_1  |   File "/usr/local/lib/python3.8/socket.py", line 796, in create_connection
celeryworker_1  |     sock.connect(sa)

**Comment by ctrung** [comment]
Hi, same problem here. 
Seems like the celeryworker container can't access the external smtp server by design. 
Not sure if the problem is adressable for non docker/python techs...

**Comment by dsever** [comment]
Guys try with:

```smtp+tls://user:pwd@server:587```

**Reply by sayanTlregtech** [reply]
Thanks that's works perfect on localhost but I am trying to implement on my hosted server and I am not able to get notifications. I am getting the error 

<html>
<body>


Email Notification | TESTING MAIL SERVICE | [Errno 99] Cannot assign requested address | Aug. 14, 2023, 6:05 a.m
-- | -- | -- | --



</body>
</html>Email Notification	[TESTING MAIL SERVICE](https://<URL>/product/22)	
[Errno 99] Cannot assign requested address

Aug. 14, 2023, 6:05 a.m

My Mailhog is running perfectly. Please help me in this situation

**Reply by Stamka** [reply]
Can you get us DD_EMAIL_URL: env variable?

**Reply by sayanTlregtech** [reply]
Sure  

DD_EMAIL: smtp+tls://<My_EMAIL>:<MY_Password>@smtp-relay.brevo.com:587

**Reply by sayanTlregtech** [reply]
This working fine in my testing server in localhost no error but in production it's not working.

**Comment by Stamka** [comment, marked as answer]
Hello everyone. 
I hope my comment will help a lot of people who struggling with email integraion. 

What did i do:
1) add environmemt vatiable to your docker-compose file with creds
**Important notice!!!
All special chars like @ ' " must be urlencoded** it was my problem


--example-- 
uwsgi:
    environment:
      DD_SESSION_COOKIE_SECURE: 'True'
      DD_CSRF_COOKIE_SECURE: 'True'
      DD_EMAIL_URL: "smtp://<email address or login with url encoded symbols. My example - tech_dojo%40domain.com>:<password with url encoded special symbols>@<smtp domain>:<ports i used 25>
  celeryworker:
    environment:
       DD_EMAIL_URL: "smtp://<email address or login with url encoded symbols. My example - tech_dojo%40domain.com>:<password with url encoded special symbols>@<smtp domain>:<ports i used 25> 
  mailhog:
    image: mailhog/mailhog:v1.0.1@sha256:8d76a3d4ffa32a3661311944007a415332c4bb855657f4f6c57996405c009bea
    entrypoint: [ "/bin/sh", "-c", "MailHog &>/dev/null" ]
    # inspired by https://github.com/mailhog/MailHog/issues/56#issuecomment-291968642
    ports:
      - target: 1025
        published: 1025
        protocol: tcp
        mode: host
      - target: 8025
        published: 8025
        protocol: tcp
        mode: host


2) do ./dc-build.sh

3) set fields in settings 

![image](https://user-images.githubusercontent.com/83986532/204315458-dac99a74-bd86-4e2a-a8ed-c1e4eadb80a7.png)


4) turn on email notifications 

![image](https://user-images.githubusercontent.com/83986532/204314975-aceeffb8-8f2b-478d-8a30-5dc98b29bd3c.png)

5) Add product or import scan. 

6) check logs 

if this answer dont help you, put here you settings and logs, i'll try to help

**Reply by sayanTlregtech** [reply]
Hey I am unable to apply the same rule for this can you help me ? I am using an smtp service sendinblue for sending the mail to my email. My defectdojo is installed into one of my virtual machine and I installed it using docker.

My smtp is :    smtp-relay.brevo.com
My port is : 587
my email is: name@companyname.com

Can you tell me where to edit that?

**Reply by sayanTlregtech** [reply]
Hey It worked on my testing vm but when I applied the same configuration the production vm it's  not working

**Comment by saeedsrm** [comment]
I still have the error
SMTPS
Port 465

error:
celeryworker-1  | Traceback (most recent call last):
celeryworker-1  |   File "/app/dojo/notifications/helper.py", line 306, in send_mail_notification
celeryworker-1  |     email.send(fail_silently=False)
celeryworker-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/mail/message.py", line 298, in send
celeryworker-1  |     return self.get_connection(fail_silently).send_messages([self])
celeryworker-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
celeryworker-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/mail/backends/smtp.py", line 124, in send_messages
celeryworker-1  |     new_conn_created = self.open()
celeryworker-1  |                        ^^^^^^^^^^^
celeryworker-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/mail/backends/smtp.py", line 80, in open
celeryworker-1  |     self.connection = self.connection_class(
celeryworker-1  |                       ^^^^^^^^^^^^^^^^^^^^^^
celeryworker-1  |   File "/usr/local/lib/python3.11/smtplib.py", line 255, in __init__
celeryworker-1  |     (code, msg) = self.connect(host, port)
celeryworker-1  |                   ^^^^^^^^^^^^^^^^^^^^^^^^
celeryworker-1  |   File "/usr/local/lib/python3.11/smtplib.py", line 343, in connect
celeryworker-1  |     (code, msg) = self.getreply()
celeryworker-1  |                   ^^^^^^^^^^^^^^^
celeryworker-1  |   File "/usr/local/lib/python3.11/smtplib.py", line 405, in getreply
celeryworker-1  |     raise SMTPServerDisconnected("Connection unexpectedly closed")
celeryworker-1  | smtplib.SMTPServerDisconnected: Connection unexpectedly closed

![image](https://github.com/user-attachments/assets/e76b4f37-7e62-488b-a803-8883b623820f)

**Reply by saeedsrm** [reply]
@Stamka

**Reply by mihaiJianu** [reply]
A solution for me was to change the volumes source and target. And now linked the whole "app" folder, for both celery worker and beat. It worked after that, maybe will help someone.
![image](https://github.com/user-attachments/assets/8d305a92-360c-4193-a4d5-778c77a73361)
